documentation/src/docs/reader-group-design.md

```diff
@@ -9,85 +9,90 @@ You may obtain a copy of the License at
+
+We still need some effective mechanism to identify, whether Readers are alive. The problem is greatly simplified because it need not produce a view of the cluster or manage any state. The component would just need to detect failures and invoke the `void ReaderOffline(String ReaderId, Position lastPosition);` API on the Reader Group.
 
 For consistency, the Failure detector should not declare a host as dead that is still processing events. Doing so could violate exactly once processing guarantees.
 
 # Examples
-## New reader
-1. When a reader joins a group its online status is added to the shared state
-1. Other readers receive updates to the shared state.
-1. When a reader with more than average number of segments sees the new reader, it may give up a segment by writing its position for that segment to the shared state.
-1. The new reader can take over a segment by recording that it is doing so in the shared state.
-1. The new reader can start reading from the position it read from the shared state for the segment it picked up.
+## New Reader
+1. When a Reader joins a group its online status is added to the shared state.
+1. Other Readers receive updates to the shared state.
+1. When a Reader with more than average number of segments sees the new Reader, it may give up a segment by writing its position for that segment to the shared state.
+1. The new Reader can take over a segment by recording that it is doing so in the shared state.
+1. The new Reader can start reading from the position it read from the shared state for the segment it picked up.
 
-There are no races between multiple readers coming online concurrently because only one of them can successfully claim ownership of any given segment.
+There are no races between multiple Readers coming online concurrently because only one of them can successfully claim ownership of any given segment.
 
 ## Segments merging
-1. When a reader comes to the end of its segment it records this information in the shared state.
-1. When all of the segments that are getting merged together are completed a reader may claim ownership of the following segment.
+1. When a Reader comes to the end of its segment it records this information in the shared state.
+1. When all of the segments that are getting merged together are completed, a Reader may claim ownership of the following segment.
+
+There is no ambiguity as to who the owner is, because it is stored in the shared state. There is no risk of a segment being ignored because every Reader can see the available segments by looking at the shared state and claim them.
+
+## Reader going offline
+1. When a Reader dies, the `readerOffline(readerId, position) api` method will be invoked either by the Reader itself in a graceful shutdown (internally to the close method) or via a "liveness detector". In either case the Reader's last position is written to the state.
```

Comment:
Maybe `api` should be API (without `this format`).

Reply:
Accepted suggestion